### PR TITLE
[eng-290] Add mempool fallback for sat hunter

### PR DESCRIPTION
**Pull request by HashTraveler**
Implemented retry logic with leverage of axios interceptors

### CODE REVIEW

**Comment by HashTraveler** [comment]
TODO: Implement also for `wallet`

File: .env.sample [hunk 1]
```diff
@@ -41,6 +41,8 @@ DEEZY_API_KEY=replace_me
 # A block explorer is needed to lookup fee rates, and for transaction and utxo lookup if bitcoin core is not used
 # Default mempool.space may get rate limited so this allows for a local server if relevant
 #MEMPOOL_URL=https://mempool.space
+#MEMPOOL_RETRY_URL=https://mempool.deezy.io
```
Comment (dannydeezy) [inline .env.sample:44]:
```suggestion
#MEMPOOL_RETRY_URL=https://replaceme.yourwebsite.com
```

File: utils/mempool.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+const axios = require('axios')
+const MEMPOOL_URL = process.env.MEMPOOL_URL || 'https://mempool.space'
+const MEMPOOL_RETRY_URL = process.env.MEMPOOL_RETRY_URL || 'https://mempool.deezy.io'
```
Comment (dannydeezy) [inline utils/mempool.js:3]:
```suggestion
const MEMPOOL_RETRY_URL = process.env.MEMPOOL_RETRY_URL || 'https://mempool-bk.deezy.io'
```
I think this is the proper one (cc @habibitcoin )?

**Comment by dannydeezy** [comment]
this isn't working for me - it hangs after a failure

**Comment by dannydeezy** [comment]
```
Attempted call with URL /api/v1/fees/mempool-blocks failed: getaddrinfo ENOTFOUND bad.deezy.io                                                                                                
Attempted 1 with URL /api/v1/fees/mempool-blocks failed: getaddrinfo ENOTFOUND bad.deezy.io
```

with this config:
```
MEMPOOL_URL="https://bad.deezy.io"
MEMPOOL_RETRY_URL="https://mempool-bk.deezy.io"
```

**Comment by HashTraveler** [comment]
> MEMPOOL_URL="https://bad.deezy.io"
> MEMPOOL_RETRY_URL="https://mempool-bk.deezy.io"

image


bad log argument, but still should work, checking..

**Comment by HashTraveler** [comment]
@dannydeezy , could you please retry?

**Comment by dannydeezy** [comment]
working now

File: .env.sample (hunk 1 above)
Comment (dannydeezy) [reply .env.sample:44]: lets fix this, then i approve

**Review by dannydeezy — APPROVED** [review]
approve pending using default of `mempool-bk.deezy.io`

**Comment by habibitcoin** [comment]
Agreed to use https://yourmempoolinstance.com/

**Review by habibitcoin — APPROVED** [review]
nice!

**Pull request merged**